### PR TITLE
API Additions and changes

### PR DESCRIPTION
**Pull request by eirikh1996**
### Describe in detail what your pull request acomplishes
In this pull request, the following is acomplished:
- CraftRotateEvent now also stores origin point and rotation
- Two new events have been added: CraftPreTranslateEvent, which is fired before the translation algorithm is executed, and ItemHarvestEvent, fired when a craft harvest blocks in its path
- Translation-related events, such as CraftPreTranslateEvent and CraftTranslateEvent now has an option added if sound should be played upon cancellation

### Checklist
- [x] Unit tests 
- [x] Proper internationalization
- [x] Compiled/tested

### CODE REVIEW

File: modules/api/src/main/java/net/countercraft/movecraft/events/ItemHarvestEvent.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.List;
+
+/**
+ * Called when one or more blocks are harvested by a moving craft
+ */
+public class ItemHarvestEvent extends CraftEvent {
+    private List<ItemStack> harvested;
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+    public ItemHarvestEvent(@NotNull Craft craft, List<ItemStack> harvested) {
```
Comment (cccm5) [inline modules/api/src/main/java/net/countercraft/movecraft/events/ItemHarvestEvent.java]: Perhaps we should also add the type of block harvested and the location (or just a block object)?

**Comment by eirikh1996** [comment]
Requested changes made @cccm5

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java [hunk 2]
```diff
@@ -302,20 +320,26 @@ private static HitBox translateHitBox(HitBox hitBox, MovecraftLocation shift){
         return output;
     }
 
-    private void fail(String message) {
+    private void fail(String failMessage) {
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java]: Use nullability anotations

**Comment by eirikh1996** [comment]
Requested changes made @cccm5

**Pull request merged**